Sdk/AssertEqualityComparer.cs

```diff
@@ -128,21 +107,21 @@ public bool Equals(T x, T y)
                 return true;
 
             // Implements IEquatable<typeof(y)>?
-            TypeInfo iequatableY = typeof(IEquatable<>).MakeGenericType(y.GetType()).GetTypeInfo();
-            if (iequatableY.IsAssignableFrom(x.GetType().GetTypeInfo()))
+            Type iequatableY = y.GetType().MakeEquatableType();
```

Comment:
@hughbe Yeah, but `typeof(IEquatable<>).MakeGenericType(...)` seems so noisy. I figured since I was refactoring some reflection-related code into a new class anyway, might as well make these callsites look prettier even though there are only 2 occurrences.